File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.py
def solution(data, ext, val_ext, sort_by):
    names = {
        'code': 0,
        'date': 1,
        'maximum': 2,
        'remain': 3,
    }
    
    answer = [d for d in data if d[names[ext]] < val_ext]
    return sorted(answer, key=lambda x: x[names[sort_by]])

File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.java
import java.util.*;

class Solution {
    
    private static final Map<String, Integer> COLUMN_MAP = Map.of(
        "code", 0,
        "date", 1,
        "maximum", 2,
        "remain", 3
    );

    public int[][] solution(int[][] data, String ext, int val_ext, String sort_by) {
        int extIdx = COLUMN_MAP.get(ext);
        int sortIdx = COLUMN_MAP.get(sort_by);

        return Arrays.stream(data)
            .filter(d -> d[extIdx] < val_ext)
            .sorted(Comparator.comparingInt(d -> d[sortIdx]))
            .toArray(int[][]::new);
    }
}